src/Http/BaseApplication.php

```diff
@@ -110,6 +118,10 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
      */
     protected function getDispatcher()
     {
-        return new ActionDispatcher(null, null, DispatcherFactory::filters());
+        if ($this->dispatcher === null) {
```

Comment:
Yes. Most plugins/apps add their dispatch filters in the bootstrap process, which is after the application is created.

How would injecting the dependency work if building in the constructor is not possible?